File: backend/app/schemas/llm.py
from app.schemas.base import CamelModel
from app.schemas.market import Direction


class LlmExplanationOut(CamelModel):
    symbol: str
    interval: str
    timestamp: int
    direction: Direction
    confidence: float
    summary: str
    key_drivers: list[str]
    risks: list[str]
    opportunities: list[str]
    assets_affected: list[str]

File: types/llm.ts
import type { Direction } from "./market";

/**
 * Mirrors the backend's LLM Explanation Layer response shape (see
 * backend/app/schemas/llm.py). `direction`/`confidence` are always copied
 * straight from the deterministic AI Decision Engine — Claude only
 * produces `summary`/`keyDrivers`/`risks`/`opportunities`/`assetsAffected`.
 */
export interface LlmExplanation {
  symbol: string;
  interval: string;
  timestamp: number;
  direction: Direction;
  confidence: number;
  summary: string;
  keyDrivers: string[];
  risks: string[];
  opportunities: string[];
  assetsAffected: string[];
}
